### ISSUE TITLE
Just pushed a trial release to CPAN

### ISSUE THREAD

**Issue by jjatria** [jlapeyre/PDL-DSP-Windows#4]
I've taken the liberty of just [pushing 0.007001 as a trial release](https://metacpan.org/release/JJATRIA/PDL-DSP-Windows-0.007001-TRIAL) to CPAN. This won't be indexed by PAUSE just yet, but it will allow the distribution to be tested by the CPAN testers.

This release is the result of a couple of weeks of work on this distribution to bring it back to a usable state. This included the fix in #3 as well as [a couple](https://github.com/jjatria/PDL-DSP-Windows/commit/a045ed04a636aa036277c4e86f39a54c8d5b7c6b) of [other fixes](https://github.com/jjatria/PDL-DSP-Windows/commit/ab1146dc53b4dfc14fb8e5c4ee0ebbcd410f36be) for [bugs](https://github.com/jjatria/PDL-DSP-Windows/commit/ababb78f2b451c177f0756dd51a049be4a4a9f97) that came up while expanding the test coverage.

Development of this release has been going on for now on [the `static` branch on my fork](https://github.com/jjatria/PDL-DSP-Windows/tree/static) (thus named because I moved to a statically generated version of the code). Part of this work included adding support for testing on [Travis CI](https://travis-ci.org/jjatria/PDL-DSP-Windows) and [Coveralls](https://coveralls.io/github/jjatria/PDL-DSP-Windows).

For now, this release marks my fork as the canonical repository for PDL::DSP::Windows (only because I do not have a commit bit on this repository). However, I'd be more than happy to turn this work into a pull request here if you prefer to keep this as the canonical repository.

All these changes respect the existing interface with one exception: **I've added a deprecation warning for the package variables that hold references to the window subroutines and window metadata, and moved the code to never read from these** (see [28d7ed](https://github.com/jjatria/PDL-DSP-Windows/commit/28d7ed524e48050d88e5d7c9a2e221bcf485a5a2)). The variables themselves are still there, and the data in them hasn't changed. However, attempting to read (or write) from them will now issue a warning like

> Package variables from PDL::DSP::Windows are deprecated and will be removed in the future. This attempt to read from them will soon become an error

I've done this because they were not necessary for what the module was doing, they were not documented (and thus not strictly part of the API), and because they made it harder to reason about what was a legitimate use of them. This required adding Variable::Magic as a dependency, but this can be removed as soon as the deprecation process is considered to be complete.

The issue was with something like

```
$ perl -Ilib -MPDL::DSP::Windows=window -E '
    $PDL::DSP::Windows::winsubs{hamming} = sub { die "Boom!" }; say window(10)'
Boom! at -e line 1.
```

which now will start to behave like

```
$ perl -Ilib -MPDL::DSP::Windows=window -E '
    $PDL::DSP::Windows::winsubs{hamming} = sub { die "Boom!" }; say window(10)'
Package variables from PDL::DSP::Windows are deprecated and will be removed in the future. This attempt to write to them will soon become an error at -e line 1.
[0.08 0.18761956 0.46012184 0.77 0.97225861 0.97225861 0.77 0.46012184 0.18761956 0.08]
```

I will welcome any feedback on any of these changes.

In the meantime, I intend to continue the experiment with this trial release on CPAN, and depending on the test results, and if no objection has been raised, to upgrade it to a proper release in a month or so.

Best!

**Comment by jjatria** [comment]
A week or so ago I've actually [released 0.007003](https://metacpan.org/release/JJATRIA/PDL-DSP-Windows-0.007003-TRIAL) as another trial release to fix a couple of failing tests (particularly with older PDLs) and it seems that has fixed things. [CPAN Testers](http://matrix.cpantesters.org/?dist=PDL-DSP-Windows%200.007003-TRIAL) currently shows 101 reports across 49 configurations, all passing.

This last release tweaks the deprecation messages a bit, to not say they will become "errors". They now say that those uses will soon "stop working". I would expect those messages to stay there for about a year after release before they are dropped entirely.

**Comment by jjatria** [comment]
These changes have now been [released to CPAN as 0.007005](https://metacpan.org/release/JJATRIA/PDL-DSP-Windows-0.007005), and the changes have been merged into the `master` branch of my fork. :tada:

In hindsight, I maybe should have bumped the version number a little more. Then again, that is easy to fix. I'll close this for now, but feel free to comment here if you'd like to talk about this some more!

**Comment by jjatria** [comment]
That's been bumped to 0.008 now. :)
